tests/document/test_document.cxx

```diff
@@ -444,51 +422,52 @@ BOOST_AUTO_TEST_CASE( save_to_file )
     doc.save_to_file(test_file.get_name());
 
     std::ifstream stream(test_file.get_name());
-    BOOST_CHECK( is_same_as_file(read_file_into_string(stream), "document/data/15.out") );
+    CHECK( is_same_as_file(read_file_into_string(stream), "document/data/15.out") );
 }
 
 
-BOOST_AUTO_TEST_CASE( save_throws_on_failure )
+TEST_CASE_METHOD( SrcdirConfig, "document/save_throws_on_failure", "[document]" )
 {
     xml::document doc(xml::node("root"));
     doc.get_root_node().push_back(xml::node(xml::node::text("invalid character: \x7")));
 
     std::string s;
-    BOOST_CHECK_THROW
-    (
-        doc.save_to_string(s),
-        xml::exception
-    );
+    REQUIRE_THROWS( doc.save_to_string(s) );
 
     temp_test_file test_file;
-    BOOST_CHECK_THROW
-    (
-        doc.save_to_file(test_file.get_name()),
-        xml::exception
-    );
+    REQUIRE_THROWS( doc.save_to_file(test_file.get_name()) );
 }
 
 
 #ifdef XMLWRAPP_USE_ZLIB
-BOOST_AUTO_TEST_CASE( save_to_file_gzip )
+TEST_CASE_METHOD( SrcdirConfig, "document/save_to_file_gzip", "[document]" )
 {
     xml::document doc(xml::node("root"));
     doc.get_root_node().push_back(xml::node("child"));
 
     temp_test_file test_file;
-    doc.save_to_file(test_file.get_name(), 9);
+    std::string gzfilename = test_file.get_name();
+    gzfilename.append(".gz");
+
+    doc.save_to_file(gzfilename.c_str(), 9);
 
     // verify that the file was can be read back as compressed
-    std::ifstream stream(test_file.get_name());
-    boost::iostreams::filtering_stream<boost::iostreams::input> filter;
-    filter.push(boost::iostreams::gzip_decompressor());
-    filter.push(stream);
-    BOOST_CHECK( is_same_as_file(read_file_into_string(filter), "document/data/15.out") );
 
-    // ...and by libxml2 directly too
-    xml::tree_parser parser(test_file.get_name());
-}
-#endif // XMLWRAPP_USE_ZLIB
+    // by libxml2 directly
+    xml::tree_parser parser(gzfilename.c_str());
 
+    // unzip and read back the uncompressed file
+    std::string command("gunzip " + gzfilename);
+
+    // suppress system() output
+#ifdef _MSC_VER
```

Comment:
This is not the right test, we need to check for OS and not the compiler (e.g. MinGW should still use `nul:` and not `/dev/null`). I.e. check for `defined(_WIN32) || defined(__WIN32__)`.

But OTOH I wonder why do we need to suppress output at all? Is it only to get rid of "gunzip not found" messages? If so, I think we can skip doing it entirely, it's not too bad to have them in addition to the warning discussed below.